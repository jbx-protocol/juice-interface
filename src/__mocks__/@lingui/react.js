const Trans = function ({ message, values }) {
    let replacedMessage = message
    for (const key in values) {
      // eslint-disable-next-line no-prototype-builtins
      if (values.hasOwnProperty(key)) {
        const regex = new RegExp(`\\{${key}\\}`, 'g')
        replacedMessage = replacedMessage.replace(regex, values[key])
      }
    }
    return replacedMessage
  },
  I18nProvider = function ({ children }) {
    return children
  }
export { Trans, I18nProvider }
