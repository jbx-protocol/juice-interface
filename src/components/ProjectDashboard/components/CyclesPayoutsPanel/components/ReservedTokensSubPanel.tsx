import {
  ArrowUpCircleIcon,
  EllipsisVerticalIcon,
} from '@heroicons/react/24/outline'
import { Trans } from '@lingui/macro'
import { Button } from 'antd'
import { twMerge } from 'tailwind-merge'
import { DisplayCard } from '../../ui'

const payouts = [
  {
    name: 'aeolian.eth',
    amount: '210 ETH',
    percent: '33%',
  },
  {
    name: 'wraeth.eth',
    amount: '210 ETH',
    percent: '33%',
  },
  {
    name: 'chris.eth',
    amount: '210 ETH',
    percent: '33%',
  },
]

export const ReservedTokensSubPanel = ({
  className,
}: {
  className?: string
}) => {
  return (
    <div className={twMerge(className)}>
      <h2 className="mb-0 font-heading text-2xl font-medium">
        <Trans>Reserved tokens</Trans>
      </h2>
      <div className="mt-5 flex flex-col items-center gap-4">
        <div className="flex w-full items-center gap-4">
          <DisplayCard className="flex w-full flex-col gap-2">
            <h3 className="text-grey-60 font-body0 mb-0 whitespace-nowrap text-sm font-medium">
              <Trans>Reserved tokens</Trans>
            </h3>
            <span className="font-heading text-xl font-medium">800,000</span>
          </DisplayCard>
          <DisplayCard className="flex w-full flex-col gap-2">
            <h3 className="text-grey-60 font-body0 mb-0 whitespace-nowrap text-sm font-medium">
              <Trans>Reserved rate</Trans>
            </h3>
            <span className="font-heading text-xl font-medium">15%</span>
          </DisplayCard>
        </div>
        <DisplayCard className="flex w-full flex-col pb-8">
          <div className="flex items-center justify-between gap-3">
            <h3 className="mb-0 whitespace-nowrap font-body text-sm font-medium">
              <Trans>Reserved tokens list</Trans>
            </h3>
            <EllipsisVerticalIcon role="button" className="h-6 w-6" />
          </div>

          <table className="mt-4 w-full">
            <tbody>
              {payouts.map(payout => (
                <tr
                  key={payout.name}
                  className="flex items-center justify-between gap-3 border-b border-grey-200 py-3"
                >
                  <td className="flex items-center gap-3">
                    <div className="h-8 w-8 rounded-full bg-juice-400" />
                    <span className="font-medium">{payout.name}</span>
                  </td>
                  <td className="flex items-center gap-3">
                    <span>{payout.amount}</span>
                    <span>{payout.percent}</span>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>

          <Button
            type="primary"
            className="mt-6 flex w-fit items-center gap-3 self-end"
          >
            <Trans>Send reserved tokens</Trans>
            <ArrowUpCircleIcon className="h-5 w-5" />
          </Button>
        </DisplayCard>
      </div>
    </div>
  )
}
