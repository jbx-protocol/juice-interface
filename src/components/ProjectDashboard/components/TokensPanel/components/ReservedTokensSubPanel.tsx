import { Trans, t } from '@lingui/macro'
import { twMerge } from 'tailwind-merge'
import { ProjectAllocationRow } from '../../ProjectAllocationRow/ProjectAllocationRow'
import { reservedTokensTooltip } from '../../TokensPanel/TokensPanelTooltips'
import { TitleDescriptionDisplayCard } from '../../ui/TitleDescriptionDisplayCard'
import { useReservedTokensSubPanel } from '../hooks/useReservedTokensSubPanel'
import { ExportTokensCsvItem } from './ExportTokensCsvItem'
import { SendReservedTokensButton } from './SendReservedTokensButton'

export const ReservedTokensSubPanel = ({
  className,
}: {
  className?: string
}) => {
  const { reservedList, reservedTokens, reservedRate } =
    useReservedTokensSubPanel()

  const reservedRateTooltip = (
    <Trans>
      {reservedRate} of token issuance is set aside for the recipients below.
    </Trans>
  )

  return (
    <div className={twMerge(className)}>
      <h2 className="mb-0 font-heading text-2xl font-medium">
        <Trans>Reserved tokens</Trans>
      </h2>
      <div className="mt-5 flex flex-col items-center gap-4">
        <div className="flex w-full flex-wrap items-center gap-4">
          <TitleDescriptionDisplayCard
            className="w-full min-w-min flex-[1_0_0]"
            title={t`Reserved tokens`}
            description={
              reservedTokens ? (
                <>{reservedTokens}</>
              ) : (
                <div className="h-7 w-24 animate-pulse rounded bg-grey-200 dark:bg-slate-200" />
              )
            }
            tooltip={reservedTokensTooltip}
          />
          <TitleDescriptionDisplayCard
            className="w-full min-w-min flex-[1_0_0]"
            title={t`Reserved rate`}
            description={reservedRate}
            tooltip={reservedRateTooltip}
          />
        </div>
        <TitleDescriptionDisplayCard
          className="w-full"
          title={t`Reserved tokens list`}
          kebabMenu={{
            items: kebabMenuItems,
          }}
        >
          {reservedList?.length ? (
            <>
              <div className="mt-4 flex w-full flex-col divide-y divide-grey-200 border-b border-grey-200 dark:divide-slate-500 dark:border-slate-500">
                {reservedList
                  ? reservedList.map(props => (
                      <ProjectAllocationRow
                        key={`${props.address}${props.projectId}`}
                        {...props}
                      />
                    ))
                  : null}
              </div>

              <SendReservedTokensButton
                className="z-0 mt-6 w-full justify-center md:w-auto"
                containerClassName="md:self-end"
              />
            </>
          ) : (
            <div className="mt-5">
              <Trans>
                No distributable reserved tokens have been configured for this
                project.
              </Trans>
            </div>
          )}
        </TitleDescriptionDisplayCard>
      </div>
    </div>
  )
}

const kebabMenuItems = [
  {
    id: 'export',
    component: <ExportTokensCsvItem />,
  },
]
