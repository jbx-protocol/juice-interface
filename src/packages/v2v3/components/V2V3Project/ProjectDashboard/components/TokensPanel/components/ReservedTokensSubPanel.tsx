import { Trans, t } from '@lingui/macro'
import { TitleDescriptionDisplayCard } from 'components/Project/ProjectTabs/TitleDescriptionDisplayCard'
import { twMerge } from 'tailwind-merge'
import { ProjectAllocationRow } from '../../ProjectAllocationRow/ProjectAllocationRow'
import { reservedTokensTooltip } from '../TokensPanelTooltips'
import { useReservedTokensSubPanel } from '../hooks/useReservedTokensSubPanel'
import { ExportTokensCsvItem } from './ExportTokensCsvItem'
import { SendReservedTokensButton } from './SendReservedTokensButton'

export const ReservedTokensSubPanel = ({
  className,
}: {
  className?: string
}) => {
  const { reservedList, reservedTokens, reservedRate } =
    useReservedTokensSubPanel()

  const reservedRateTooltip = (
    <Trans>
      {reservedRate} of token issuance is set aside for the recipients below.
    </Trans>
  )

  return (
    <div className={twMerge(className)}>
      <h2 className="mb-0 font-heading text-2xl font-medium">
        <Trans>Reserved tokens</Trans>
      </h2>
      <div className="mt-5 flex flex-col items-center gap-4">
        <div className="flex w-full flex-wrap items-center gap-4">
          <TitleDescriptionDisplayCard
            className="w-full min-w-min flex-[1_0_0]"
            title={t`Reserved tokens`}
            description={
              reservedTokens ? (
                <>{reservedTokens}</>
              ) : (
                <div className="h-7 w-24 animate-pulse rounded bg-grey-200 dark:bg-slate-200" />
              )
            }
            tooltip={reservedTokensTooltip}
          />
          <TitleDescriptionDisplayCard
            className="w-full min-w-min flex-[1_0_0]"
            title={t`Reserved rate`}
            description={reservedRate}
            tooltip={reservedRateTooltip}
          />
        </div>
        {reservedRate &&
        reservedTokens &&
        reservedTokens !== '0' &&
        reservedRate !== '0' ? (
          <TitleDescriptionDisplayCard
            className="w-full"
            title={t`Reserved tokens list`}
            kebabMenu={{
              items: kebabMenuItems,
            }}
          >
            {reservedTokens ||
            reservedRate ||
            (reservedList && reservedList.length > 1) ? (
              <>
                <div className="mt-4 flex w-full flex-col divide-y divide-grey-200 border-b border-grey-200 dark:divide-slate-500 dark:border-slate-500">
                  {reservedList
                    ? reservedList.map(props => (
                        <ProjectAllocationRow
                          key={`${props.address}${props.projectId}`}
                          {...props}
                        />
                      ))
                    : null}
                </div>

                <SendReservedTokensButton
                  className="z-0 w-full justify-center md:w-auto"
                  containerClassName="md:self-end mt-6 inline-flex"
                />
              </>
            ) : (
              <div className="mt-5">
                <Trans>
                  No distributable reserved tokens have been configured for this
                  project.
                </Trans>
              </div>
            )}
          </TitleDescriptionDisplayCard>
        ) : null}
      </div>
    </div>
  )
}

const kebabMenuItems = [
  {
    id: 'export',
    component: <ExportTokensCsvItem />,
  },
]
