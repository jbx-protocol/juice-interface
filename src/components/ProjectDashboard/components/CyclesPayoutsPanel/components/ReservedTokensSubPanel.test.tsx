/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react'
import { ReservedTokensSubPanel } from './ReservedTokensSubPanel'

describe('ReservedTokensSubPanel', () => {
  it('renders', () => {
    render(<ReservedTokensSubPanel />)
  })
})
