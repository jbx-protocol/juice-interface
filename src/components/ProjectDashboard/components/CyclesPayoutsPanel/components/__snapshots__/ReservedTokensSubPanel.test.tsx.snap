// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservedTokensSubPanel renders 1`] = `
<div>
  <div
    class=""
  >
    <h2
      class="mb-0 font-heading text-2xl font-medium"
    >
      Reserved tokens
    </h2>
    <div
      class="mt-5 flex flex-col items-center gap-4"
    >
      <div
        class="flex w-full items-center gap-4"
      >
        <div
          class="rounded-lg bg-smoke-50 py-5 px-6 dark:bg-slate-700 flex w-full flex-col gap-2"
        >
          <h3
            class="text-grey-60 font-body0 mb-0 whitespace-nowrap text-sm font-medium dark:text-slate-200"
          >
            Reserved tokens
          </h3>
          <span
            class="font-heading text-xl font-medium"
          >
            21%
          </span>
        </div>
        <div
          class="rounded-lg bg-smoke-50 py-5 px-6 dark:bg-slate-700 flex w-full flex-col gap-2"
        >
          <h3
            class="text-grey-60 font-body0 mb-0 whitespace-nowrap text-sm font-medium dark:text-slate-200"
          >
            Reserved rate
          </h3>
          <span
            class="font-heading text-xl font-medium"
          >
            82%
          </span>
        </div>
      </div>
      <div
        class="rounded-lg bg-smoke-50 py-5 px-6 dark:bg-slate-700 flex w-full flex-col pb-8"
      >
        <div
          class="flex items-center justify-between gap-3"
        >
          <h3
            class="mb-0 whitespace-nowrap font-body text-sm font-medium dark:text-slate-200"
          >
            Reserved tokens list
          </h3>
          <svg
            aria-hidden="true"
            class="h-6 w-6"
            fill="none"
            role="button"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 6.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 12.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 18.75a.75.75 0 110-1.5.75.75 0 010 1.5z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="mt-4 flex w-full flex-col divide-y divide-grey-200 border-b border-grey-200 dark:divide-slate-500 dark:border-slate-500"
        >
          <div>
            {"address":"0x00000","percent":"0%"}
          </div>
          <div>
            {"address":"0x00001","percent":"21%","projectId":1}
          </div>
        </div>
        <div>
          {"className":"mt-6 self-end"}
        </div>
      </div>
    </div>
  </div>
</div>
`;
